### ISSUE TITLE
Workstation 23.2.1028 - knife ssl - uninitialized class variable

### ISSUE THREAD

**Issue by TimothyTitan** [chef/chef-workstation#3072]
## Description
`knife ssl check` and `knife ssl fetch` are both returning the following error:
```
/opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/proxifier-1.0.3/lib/uri/socks.rb:8:in `<module:URI>': uninitialized class variable @@schemes in URI (NameError)
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/proxifier-1.0.3/lib/uri/socks.rb:3:in `<top (required)>'
	from <internal:/opt/chef-workstation/embedded/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/opt/chef-workstation/embedded/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/proxifier-1.0.3/lib/proxifier.rb:2:in `<top (required)>'
	from <internal:/opt/chef-workstation/embedded/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/opt/chef-workstation/embedded/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/chef-18.0.185/lib/chef/mixin/proxified_socket.rb:18:in `<top (required)>'
	from <internal:/opt/chef-workstation/embedded/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from <internal:/opt/chef-workstation/embedded/lib/ruby/3.1.0/rubygems/core_ext/kernel_require.rb>:85:in `require'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/knife-18.0.185/lib/chef/knife/ssl_check.rb:33:in `block in <class:SslCheck>'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/knife-18.0.185/lib/chef/knife.rb:245:in `each'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/knife-18.0.185/lib/chef/knife.rb:245:in `load_deps'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/knife-18.0.185/lib/chef/knife.rb:230:in `run'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/knife-18.0.185/lib/chef/application/knife.rb:165:in `run'
	from /opt/chef-workstation/embedded/lib/ruby/gems/3.1.0/gems/knife-18.0.185/bin/knife:24:in `<top (required)>'
	from /usr/local/bin/knife:400:in `load'
	from /usr/local/bin/knife:400:in `<main>'
```

## Chef Workstation Version
Chef Workstation version: 23.2.1028
Test Kitchen version: 3.5.0
Cookstyle version: 7.32.1
Chef Infra Client version: 18.0.185
Chef InSpec version: 5.18.14
Chef CLI version: 5.6.8
Chef Habitat version: 1.6.521

Also tested in the 23.3.1030 development version with the same error

## Platform Version
We built 23.2.1028 via omnibus on an intel mac os architecture and noticed knife ssl commands were failing. To validate it wasn't an issue in our build process - we downloaded the current stable and development releases from chef.io on a windows machine and experienced the same errors in both check and fetch.
Tested on macos 12.6, macos 13.0 and Windows 10 Pro (10.0.19045)

**Comment by vkarve-chef** [comment]
we have a [fix](https://github.com/chef/chef/pull/13617) in place. once the updated chef gem is published, we will have a followup chef-workstation release so this issue get addressed.

**Comment by red8888** [comment]
- @vkarve-chef does downgrading fix this? What version of workstation is working? Im on `Chef Infra Client: 18.0.185`

**Comment by red8888** [comment]
I trying to download a previous version but all previous version links on the website are dead

**Comment by jugal510** [comment]
i have the same error.

**Comment by vkarve-chef** [comment]
@red8888  Chef Workstation 22.12.1024 or lower will work alright for these knife commands.

**Comment by vkarve-chef** [comment]
the issue is fixed in latest release 23.4.1032
